dfget/core/downloader/p2p_downloader/p2p_downloader.go

```diff
@@ -228,6 +230,9 @@ func (p2p *P2PDownloader) pullPieceTask(item *Piece) (
 		if res, err = p2p.API.PullPieceTask(item.SuperNode, req); err != nil {
```

Comment:
Could you please help to refactor the code here a little bit? @wybzju I think we could refactor that in this way:
```
for {
    res, err = p2p.API.PullPieceTask(item.SuperNode, req); 
    if err != nil {
        logrus.Errorf("pull piece task error: %v", err)
        break
    }
    if res.Code != constants.CodePeerWait {
        break
    }
    if p2p.queue.Len() > 0 {
        break
    }
    sleepTime := time.Duration(rand.Intn(1400)+600) * time.Millisecond
    logrus.Infof("pull piece task result:%s and sleep %.3fs", res, sleepTime.Seconds())
    time.Sleep(sleepTime)
}
```

I think that could be more readable. And the logic is the same. Please strictly follow https://github.com/alibaba/pouch/blob/master/docs/contributions/code_styles.md#rule006---return-fast-to-indent-less to improve that. 
/cc @Starnop @lowzj @yeya24

Reply:
Done

Reply:
Done